src/devices/rojaflex.c

```diff
@@ -0,0 +1,317 @@
+        uint16_t crc_calc    = crc16(&msg[LENGTH_OFFSET], 9, 0x8005, 0xffff);
+
+        if (crc_message != crc_calc) {
+            if (decoder->verbose)
+                fprintf(stderr, "%s: CRC invalid message:%04x != calc:%04x\n", __func__, crc_message, crc_calc);
+
+            return DECODE_FAIL_MIC;
+        };
+    }
+
+    // Build the default terminal output
+    {
+        char tokenString[7] = "";
+        char id[10]         = "";
+        char *deviceType    = "unknown";
+        sprintf(tokenString, "0x%02x%02x", msg[MESSAGE_TOKEN_OFFSET], msg[MESSAGE_TOKEN_OFFSET + 1]);
+        sprintf(id, "0x%02x%02x%02x%01x", msg[ID_OFFSET], msg[ID_OFFSET + 1], msg[ID_OFFSET + 2], (msg[ID_OFFSET + 3] >> 4));
+
+        if ((msg[COMMAND_ID_OFFSET] & 0xF) == 0x5) {
+            deviceType = "RojaFlex-Shutter";
+        }
+        else if ((msg[COMMAND_ID_OFFSET] & 0xF) == 0xa) {
+            // Rojaflex Bridge clones a remote signal but does not send an CRC!?!?
+            // So we can detect if it a real remote or a bridge on the message length.
+            if (dataframe_bitcount == DATAFRAME_BITCOUNT_INCL_CRC) {
+                deviceType = "RojaFlex-Remote";
+            }
+            else {
+                deviceType = "RojaFlex-Bridge";
+            }
+        }
+
+        /* clang-format off */
+        data = data_make(
+                    "model",        "Model",     DATA_STRING, deviceType,
+                    "id",           "ID",        DATA_STRING, id,
+                    "channel",      "Channel",   DATA_INT,    msg[CHANNEL_OFFSET] & 0xF,
+                    "token",        "Msg Token", DATA_STRING, tokenString,
+                    "commandtype",  "Command ",  DATA_STRING, getCommandString(&msg[0]),
+                    "commandvalue", "Value",     DATA_INT,    msg[COMMAND_VALUE_OFFSET],
+                    "mic",          "Integrity", DATA_STRING, (dataframe_bitcount == DATAFRAME_BITCOUNT_INCL_CRC) ? "CRC" : "",
+                    NULL);
+        /* clang-format on */
+
+        decoder_output_data(decoder, data);
+    }
+
+// You can use this defines to clone / generate all commands for other bridges
+#define GENERATE_COMMANDS_FOR_CURRENT_CHANNEL 0
+#define GENERATE_COMMANDS_FOR_ALL_CHANNELS 0
+
+    if (GENERATE_COMMANDS_FOR_CURRENT_CHANNEL || GENERATE_COMMANDS_FOR_ALL_CHANNELS) {
+        uint8_t const remote_commands[] = {
+                COMMAND_ID_STOP,
+                COMMAND_ID_UP,
+                COMMAND_ID_DOWN,
+                COMMAND_ID_SAVE_UNSAVE_POS,
+                COMMAND_ID_GO_SAVED_POS,
+                COMMAND_ID_REQUESTSTATUS};
+
+        uint8_t channel = GENERATE_COMMANDS_FOR_CURRENT_CHANNEL ? msg[CHANNEL_OFFSET] & 0xF : 0;
+        uint8_t command = 0;
+        uint8_t msg_new[19];
+        uint16_t sum = 0;
+
+        fprintf(stderr, "\n");
+        fprintf(stderr, "%s: Signal cloner\n", __func__);
+        fprintf(stderr, "%s: \n", __func__);
```

Comment:
printouts need to be around verbose guards.

Here you can see how to do it:
https://github.com/merbanan/rtl_433/blob/master/src/devices/ambientweather_tx8300.c#L69

Reply:
done

I did not do this before, because this is special code, which you need to activate by a define. This will not happen during normal usage.